src/sources/soundsourcemp3.cpp

```diff
@@ -276,6 +288,40 @@ SoundSource::OpenResult SoundSourceMp3::tryOpen(
             continue;
         }
 
+        if (!mp3InfoTagSkipped) {
+            // We assume that the first frame contains the mp3 info frame
+            // which needs to be skipped
+            // https://linux.m2osw.com/mp3-info-tag-specifications-rev0-lame-3100
+            // https://github.com/Iunusov/libmp3lame-CMAKE/blob/bb770fb6e4b4dfc963860380a5e7765c370aaef1/libmp3lame/VbrTag.c#L333
+            int mp3InfoFrameOffset = 0;
+            if (madHeader.flags & MAD_FLAG_LSF_EXT) {
+                // MPEG Version 2 (ISO/IEC 13818-3)
+                // or MPEG Version 2.5
+                if (madHeader.mode == MAD_MODE_SINGLE_CHANNEL) {
+                    mp3InfoFrameOffset = (9 + 4);
+                } else {
+                    mp3InfoFrameOffset = (17 + 4);
+                }
+            } else {
+                // MPEG Version 1 (ISO/IEC 11172-3)
+                if (madHeader.mode == MAD_MODE_SINGLE_CHANNEL) {
+                    mp3InfoFrameOffset = (17 + 4);
+                } else {
+                    mp3InfoFrameOffset = (32 + 4);
+                }
+            }
+
+            QString mp3InfoLabel =
+                    QString::fromLatin1(reinterpret_cast<const char*>(
+                            &m_madStream.this_frame[mp3InfoFrameOffset]));
+            kLogger.debug()
+                    << "Skipping MP3 Info Frame:"
+                    << mp3InfoLabel;
```

Comment:
Can we always be sure that the tagname is null-terminated?

Reply:
No, good catch. Fixed.